packages/nodes-base/nodes/Splunk/v2/helpers/utils.ts

```diff
@@ -0,0 +1,106 @@
+import type { IExecuteFunctions, IDataObject } from 'n8n-workflow';
+
+import { parseString } from 'xml2js';
+
+import type { SplunkError, SplunkFeedResponse } from './interfaces';
+
+function compactEntryContent(splunkObject: any): any {
+	if (typeof splunkObject !== 'object') {
+		return {};
+	}
+
+	if (Array.isArray(splunkObject)) {
+		return splunkObject.reduce((acc, cur) => {
+			acc = { ...acc, ...compactEntryContent(cur) };
+			return acc;
+		}, {});
+	}
+
+	if (splunkObject['s:dict']) {
```

Comment:
suggestion: avoid using magic strings 's:dict' is used in many places, to enhance maintainability, readability, and reduce the risk of errors, consider defining it as a constant.